madara/crates/client/gateway/client/src/methods.rs

```diff
@@ -36,6 +37,16 @@ impl GatewayProvider {
         }
     }
 
+    pub async fn get_block_with_header_only(&self) -> Result<ProviderBlockHeaderOnly, SequencerError> {
```

Comment:
this endpoint needs some tests but
turns out you're in luck and you don't have to write them, because I also added this endpoint in #495 and wrote some tests.